### PR TITLE
Poseidon BIG update

### PR DESCRIPTION
**Pull request by ya4ept**
Trying to fix Poseidon for twRO, vRO, cRO, jRO
- [x] minimum twRP support
- [x] minimum vRO support
- [x] minimum cRO support
- [x] minimum jRO support
- [x] minimum kRO support
- [x] minimum kRO Zero support
- [x] Dev mode do not responding to GG request
- [x] Dev mode does not show NPCs and Poring
- [x] start-poseidon.exe takes no arguments: `start-poseidon.exe --file=test.txt`
- [x] update Poseidon version

### CODE REVIEW

File: src/Poseidon/RagnarokServer.pm [hunk 1]
```diff
@@ -1002,24 +1005,24 @@ sub SendCharacterList
...
 	} else {
-		if ( $config{server_type} !~ /^bRO/ ) { $data .= $accountID; } #<- This is Server Type Based !!
-		$data = pack("C*", 0x73, 0x00) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05);
-		$client->send($data);
+		# '0073' => ['map_loaded','x4 a3',[qw(coords)]]
+		$client->send(pack("v", 0x0073) . pack("V", getTickCount) . getCoordString($posX, $posY, 1) . pack("C*", 0x05, 0x05);
 	}
 
+	if ( $config{server_type} =~ /^kRO/ ) { # kRO
+		$client->send(pack("C*", 0xDE, 0x0A, 0x46, 0x00, 0x00, 0x00)); # Overweight percent
```
Comment (ya4ept) [inline src/Poseidon/RagnarokServer.pm:1023]:
```c++
struct PACKET_ZC_OVERWEIGHT_PERCENT {
	int16 packetType;
	uint32 percent;
} __attribute__((packed));
```
need to be replaced in `Sakexe_0` and `ServerType0`
~'0ADE' => ['overweight_percent', 'v V', [qw(len percent)]],#TODO~
```perl
#'0ADE' => ['overweight_percent', 'V', [qw(percent)]] # 6 TODO
```
RagnarokServer.pm:
~$client->send(pack("C*", 0xDE, 0x0A, 0x46, 0x00, 0x00, 0x00)); # Overweight percent~
```perl
$client->send(pack("v", 0x0ADE) . pack("V", 0x00);
```

Comment (alisonrag) [reply src/Poseidon/RagnarokServer.pm]: fixed at https://github.com/OpenKore/openkore/pull/3468/commits/96e2fb6443ccade23155da5d35ba5d1cfc421bbf

File: control/poseidon.txt [hunk 2]
```diff
@@ -15,7 +15,7 @@ queryserver_port=24390
 # to the poseidon operate properly !
 # The available server types for now are : Default, bRO_.* (check servertypes.txt)
 # You should modify this if you're having problems with char list.
-server_type=bRO_2016-11-08a
+server_type=Default
 
 # Char Block Size
```
Comment (ya4ept) [inline control/poseidon.txt:20]: i think we should remove this. This parameter is not used anywhere

Comment (alisonrag) [reply control/poseidon.txt]: fixed at https://github.com/OpenKore/openkore/pull/3468/commits/8b056b89b8e962d7c7c4bba7b1140705989af158

**Comment by ya4ept** [comment]
if you enable debug in poseidon and write a message to the general chat in the client, then two identical messages are displayed in the console
image

**Comment by alisonrag** [comment]
> if you enable debug in poseidon and write a message to the general chat in the client, then two identical messages are displayed in the console
> image

this is because of double visualDump
1 - when receive packet if debug = 1
2 - when unknown packet and char = 'poseidon dev'

**Comment by ya4ept** [comment]
let's change the version of poseidon
`Starting Poseidon 2.1`

**Comment by ya4ept** [comment]
@alisonrag, let's debug outgoing messages? Sometimes it's helpful to see exactly what a poseidon is answering.

**Comment by alisonrag** [comment]
> @alisonrag, let's debug outgoing messages? Sometimes it's helpful to see exactly what a poseidon is answering.
[added](https://github.com/OpenKore/openkore/pull/3468/commits/aff3ff2557b3ccc2bd6ec370e3482f9c264621b8)
image

**Comment by alisonrag** [comment]
> * [ ]  start-poseidon.exe takes no arguments: `start-poseidon.exe --file=test.txt`
if i'm not wrong `poseidon.pl` was updated here to support command line arguments:
https://github.com/OpenKore/openkore/pull/1273
and the start-poseidon.exe was not rebuild using this new `poseidon.pl`

**Comment by alisonrag** [comment]
> * [x]  minimum jRO support

[added](https://github.com/OpenKore/openkore/pull/3468/commits/6f16fd9c713d7c1803d0a08741068db9b12a9c08)

**Comment by ya4ept** [comment]
> > * [ ]   start-poseidon.exe takes no arguments: `start-poseidon.exe --file=test.txt`
> 
> if i'm not wrong `poseidon.pl` was updated here to support command line arguments:
> #1273
> and the start-poseidon.exe was not rebuild using this new `poseidon.pl`

You're wrong. `start-poseidon.exe` - does not contain any data from `poseidon.pl`, this is just a link to the file `src\Poseidon\poseidon.pl`. It does not need to be generated every time poseidon.pl changes.

I think we have a bug somewhere inside the "script-launcher" logic. It is not designed for passing command line arguments.

**Comment by timelinebk9x** [comment]
Excuse me if my comment was at wrong place. I have an issue when connecting vRO client to Poseidon server. The client connected to the server but could not load character list, it got stuck at waiting dialogue and disconnected after 1 minute. Is there any similar issue and guide to solve this?

**Comment by ya4ept** [comment]
> Dev mode do not responding to GG request
this works fine (checked on twRO)
let's merge

**Review by alisonrag — APPROVED** [review]

**Pull request merged**